contracts/pair_stable/src/contract.rs

```diff
@@ -496,8 +496,27 @@ pub fn provide_liquidity(
     let deposit_d = compute_d(amp, &new_balances, config.greatest_precision)?;
 
     let total_share = query_supply(&deps.querier, &config.pair_info.liquidity_token)?;
-    let mint_amount = if total_share.is_zero() {
-        deposit_d
+    let share = if total_share.is_zero() {
+        let share = deposit_d
+            .to_uint128_with_precision(config.greatest_precision)?
+            .checked_sub(MINIMUM_LIQUIDITY_AMOUNT)
+            .map_err(|_| ContractError::MinimumLiquidityAmountError {})?;
+
+        messages.extend(mint_liquidity_token_message(
+            deps.querier,
+            &config,
+            &env.contract.address,
+            &env.contract.address,
+            MINIMUM_LIQUIDITY_AMOUNT,
+            false,
+        )?);
+
+        // share cannot become zero after minimum liquidity subtraction
+        if share.is_zero() {
```

Comment:
This check is repeated in else-branch. I suggest to move it outside the if-else clause (499-557).

Reply:
That's due to different error messages

Reply:
Ah, ok